hermes/src/network/p2p.rs

```diff
@@ -72,28 +72,39 @@ lazy_static::lazy_static! {
 /// This function is passed as a callback to the Go libp2p runtime, it passes observations back and
 /// acts as a proxy forwarding these observations into our main loop.
 #[no_mangle]
+#[tracing::instrument(skip(o))]
 extern "C" fn proxy(o: ObservationC) {
     // Create a fixed slice from the pointer and length.
     let vaa = unsafe { std::slice::from_raw_parts(o.vaa, o.vaa_len) }.to_owned();
-    // The chances of the mutex getting poisioned is very low and if it happens
-    // there is no way for us to recover from it.
+
+    let hash = &{
+        let mut bytes = <sha3::Sha3_256 as sha3::Digest>::new();
+        sha3::Digest::update(&mut bytes, &vaa);
+        hex::encode(sha3::Digest::finalize(bytes))
+    };
```

Comment:
what is this used for?

Reply:
It's associated with the Span that's stored with the VAA, so all logs related to this VAA are tagged with its hash, makes it easy to filter logs by VAA to get a focused trace on events related to this VAA as it moves through the program. Could probably use sequence number instead I just wanted to see for sure that duplicate VAAs were handled right.